examples/virtual_methods/base_button/imp.rs

```diff
@@ -0,0 +1,103 @@
+use super::BaseButtonExt;
+use gtk::{
+    gio,
+    glib::{self, Error},
+    prelude::*,
+    subclass::prelude::*,
+};
+use std::{future::Future, pin::Pin};
+
+pub type BaseButtonInstance = super::BaseButton;
+pub type PinnedFuture = Pin<Box<dyn Future<Output = Result<(), Error>> + 'static>>;
+
+/// GObject glue code for our BaseButtonClass which holds the function pointers to our virtual functions.
+#[repr(C)]
+pub struct BaseButtonClass {
+    pub parent_class: gtk::ffi::GtkButtonClass,
+    pub sync_method: Option<unsafe fn(&BaseButtonInstance, extra_text: Option<String>)>,
+    pub async_method: Option<unsafe fn(&BaseButtonInstance) -> PinnedFuture>,
+}
+
+unsafe impl ClassStruct for BaseButtonClass {
+    type Type = BaseButton;
+}
+
+impl std::ops::Deref for BaseButtonClass {
+    type Target = glib::Class<glib::Object>;
+
+    fn deref(&self) -> &Self::Target {
+        unsafe { &*(self as *const _ as *const Self::Target) }
+    }
+}
+
+impl std::ops::DerefMut for BaseButtonClass {
+    fn deref_mut(&mut self) -> &mut glib::Class<glib::Object> {
+        unsafe { &mut *(self as *mut _ as *mut glib::Class<glib::Object>) }
+    }
+}
+
+#[derive(Debug, Default)]
+pub struct BaseButton;
+
+// Virtual method default implementation trampolines
+unsafe fn sync_method_default_trampoline(this: &BaseButtonInstance, extra_text: Option<String>) {
+    BaseButton::from_instance(this).sync_method(this, extra_text)
+}
+
+unsafe fn async_method_default_trampoline(this: &BaseButtonInstance) -> PinnedFuture {
+    BaseButton::from_instance(this).async_method(this)
+}
+
+pub unsafe fn base_button_sync_method(this: &BaseButtonInstance, extra_text: Option<String>) {
+    let klass = &*(this.class() as *const _ as *const BaseButtonClass);
+    (klass.sync_method.unwrap())(this, extra_text)
+}
+
+pub unsafe fn base_button_async_method(this: &BaseButtonInstance) -> PinnedFuture {
```

Comment:
```suggestion
pub(super) unsafe fn base_button_async_method(this: &BaseButtonInstance) -> PinnedFuture {
```

maybe. To limit visibility to the parent module. Also for the other one above.